examples/middleware_concurrency_control.rb

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env ruby
+
+example_description = <<DESC
+  Middleware Concurrency Control Example
+  ======================================
+
+  This example shows, how ConcurrencyControl middleware
+  can control concurrency level and distribute tasks over time.
+
+  This is useful, when doing resource-intensive bulk actions,
+  where running all actions at once would drain the system's resources.
+
+DESC
+
+require_relative 'example_helper'
+
+class CostyAction < Dynflow::Action
+
+  SleepTime = 10
+
+  def plan(number)
+    action_logger.info("#{number} PLAN")
+    plan_self(:number => number)
+  end
+
+  def run(event = nil)
+    unless output.key? :slept
+      output[:slept] = true
+      suspend do |suspended_action|
+        action_logger.info("#{input[:number]} SLEEP")
+        world.clock.ping(suspended_action, SleepTime)
+      end
+    end
+  end
+
+  def finalize
+    action_logger.info("#{input[:number]} DONE")
+  end
+end
+
+
+class ConcurrencyControlExample < Dynflow::Action
+  include Dynflow::Action::WithSubPlans
+
+  middleware.use ::Dynflow::Middleware::Common::ConcurrencyControl
+
+  ConcurrencyLevel = 2
+  RunWithin = 2 * 60
+  
+  def plan(count)
+    limit_concurrency_level(ConcurrencyLevel)
+    distribute_over_time(RunWithin, count, ConcurrencyLevel)
+    super(:count => count)
+  end
+
+  def create_sub_plans
+    sleep 1
```

Comment:
It is just a hacky workaround to have the plans' outputs printed after the description and webrick initialization notices